PxGraf/Models/SavedQueries/ArchiveCube.cs

```diff
@@ -1,105 +1,87 @@
         /// Collection of key (variable value code coordinates) value (double or missing code) pairs.
         /// </summary>
-        [JsonProperty("data")]
-        public List<double?> Data { get; set; }
+        [JsonConverter(typeof(DecimalDataValueListConverter))]
+        public List<DecimalDataValue> Data { get; }
 
         /// <summary>
         /// Notes mapped to the indexes of the data array.
         /// </summary>
-        [JsonProperty("dataNotes")]
-        public Dictionary<int, string> DataNotes { get; set; }
+        public Dictionary<int, MultilanguageString> DataNotes { get; }
 
         /// <summary>
-        /// Parameterles constructor for json deserialization.
+        /// Verison of the archive cube.
         /// </summary>
-        public ArchiveCube() { }
+        public string Version { get; }
 
-        private ArchiveCube(DataCube cube)
+        private ArchiveCube(IReadOnlyMatrixMetadata meta, IReadOnlyList<DecimalDataValue> data)
         {
             CreationTime = DateTime.Now;
+            Meta = meta;
 
-            Meta = cube.Meta.Clone();
-
-            List<double?> data = [];
-            Dictionary<int, string> dataNotes = [];
-
-            DataValue[] dataList = cube.Data;
-            int dataLen = dataList.Length;
-
-            for (int i = 0; i < dataLen; i++)
-            {
-                if (dataList[i].Type == DataValueType.Exist)
-                {
-                    data.Add(dataList[i].UnsafeValue);
-                }
-                else
-                {
-                    data.Add(null);
-                    dataNotes[i] = dataList[i].ToMachineReadableString(0);
-                }
-            }
-
-            Data = data;
-            DataNotes = dataNotes;
+            Data = [.. data];
+            DataNotes = [];
         }
 
         /// <summary>
         /// Creates a data cube object based on the archive cube.
         /// </summary>
         /// <returns></returns>
-        public DataCube ToDataCube()
+        public Matrix<DecimalDataValue> ToMatrix()
         {
-            DataValue[] newData = new DataValue[Data.Count];
-            int dataLen = Data.Count;
-            for (int i = 0; i < dataLen; i++)
-            {
-                if (Data[i].HasValue)
-                {
-                    newData[i] = DataValue.FromRaw(Data[i].Value);
-                }
-                else
-                {
-                    newData[i] = DataValue.FromDotCode(DataNotes[i], DataValueType.Missing);
-                }
-            }
+            Matrix<DecimalDataValue> matrix = new(Meta, [..Data]);
+            matrix.ApplyDataNotesToMissingData(DataNotes);
+            return matrix;
+        }
 
-            return new DataCube(Meta, newData);
+        public static ArchiveCube FromMatrixAndQuery(Matrix<DecimalDataValue> matrix, MatrixQuery query)
+        {
+            IReadOnlyMatrixMetadata queriedMeta = matrix.Metadata.FilterDimensionValues(query);
```

Comment:
This should not be needed and even if it did, this would break the connection between the data and metadata.

Reply:
This method has been removed.